src/modules/high_cpu.rs

```
@@ -0,0 +1,94 @@
+use super::Module;
+use crate::colors::{Color, Style};
+use crate::icons::Icon;
+use std::collections::HashMap;
+use std::fmt;
+use std::process::Command;
+
+/// A module to display the process with the highest CPU usage.
+///
+/// NOTE that the high CPU process is determined by the `ps` command.
+/// So, this is only available on linux or macos.
+pub struct HighCpu {
+    /// Field to store process information
+    process_info: String,
+}
+
+impl fmt::Display for HighCpu {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.process_info)
+    }
+}
+
+impl Default for HighCpu {
+    fn default() -> Self {
+        Self {
+            process_info: "".to_string(),
+        }
+    }
+}
+
+impl HighCpu {
+    /// Creates a new instance of HighCpuModule.
+    ///
+    /// This method runs the `ps` command to get the process information,
+    /// processes the output to find the process with the highest CPU usage,
+    /// and returns a boxed Module containing the HighCpuModule.
+    ///
+    /// Note, that this is only available on linux or macos.
+    /// When the "ps" command fails, it returns a default Module.
+    ///
+    /// # Returns
+    ///
+    /// A `Box<Module<HighCpuModule>>` containing the process information.
+    pub fn new(fg: Option<Color>, bg: Option<Color>) -> Box<Module<HighCpu>> {
```

Instead of having the errors as the process info this should return a `Result`. Then the caller can decide how to handle the errors, maybe they want to show an error text with a different style, maybe they want to ignore the error...